crates/wasmtime/src/runtime.rs

```diff
@@ -904,6 +916,40 @@ impl Store {
         }
     }
 
+    pub(crate) fn register_stack_maps(&self, module: &Module) {
+        if module
+            .compiled_module()
+            .finished_functions()
+            .iter()
+            .next()
+            .map_or(true, |(_index, f)| {
+                self.stack_map_registry()
+                    .lookup_stack_map(unsafe { (**f).as_ptr() as usize })
+                    .is_some()
+            })
+        {
+            // Either there aren't any functions in the module, or we already
+            // registered it in this store.
```

Comment:
To confirm, is every function guaranteed to have a stack map, even if it doesn't use reference types?

Reply:
Good eye, this is a bug introduced during all the refactoring today. I'm going to just make `StackMapRegistry::register_stack_maps` idempotent, rather than trying to check if its already registered here. This is a nice little clean up, since the registry is in the best position to answer this question anyways.